spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -272,6 +273,10 @@ protected void configurePollingEndpoint(AbstractPollingEndpoint pollingEndpoint,
 			String fixedDelayValue = this.environment.resolvePlaceholders(poller.fixedDelay());
 			String fixedRateValue = this.environment.resolvePlaceholders(poller.fixedRate());
 			String maxMessagesPerPollValue = this.environment.resolvePlaceholders(poller.maxMessagesPerPoll());
+			if (pollingEndpoint instanceof SourcePollingChannelAdapter) {
+				// SPCAs default to 1 message per poll
+				maxMessagesPerPollValue = "1";
```

Comment:
This is a bug. You just override any value from the annotation.
We need do that only when `!StringUtils.hasText(maxMessagesPerPollValue)` and yes when it is `SourcePollingChannelAdapter`.
Fixing during merge with appropriate test to be sure that the value isn't overridden